openhands/runtime/client/client.py

```diff
@@ -467,6 +478,48 @@ async def write(self, action: FileWriteAction) -> Observation:
             return ErrorObservation(f'Malformed paths not permitted: {filepath}')
         return FileWriteObservation(content='', path=filepath)
 
+    async def edit(self, action: FileEditAction) -> Observation:
+        diff_blocks = re.search(
+            f'(.*)\n{HEAD}(.*)\n{DIVIDER}(.*)\n{TAIL}', action.diff_block, re.DOTALL
```

Comment:
Idea for improvement: if HEAD and/or DIVIDER are found, but not TAIL, it would mean that the output got cut off.
In that case we might pass a more informative error back to the LLM instead of just "Could not resolve...", because if I'm not mistaken, the LLM does not actually _see_ that output gets cut off, or does it? 🤔